resources/startup.sh

```diff
@@ -112,131 +103,104 @@ function writeProxyCredentialsTo(){
-    if sql "SELECT 1 FROM schema_migrations WHERE version='1153';" &> /dev/null; then
-      break
-    else
-      N=$[$N+1]
-      sleep 10
-    fi
-  done
+  echo "Waiting for SonarQube status endpoint to be available (max. 120 seconds)..."
+  wait_for_sonar_status_endpoint 120
 
-  # waiting for sonar to get healthy
-  if ! doguctl wait-for-http --timeout 120 --method GET http://localhost:9000/sonar/api/system/status; then
-    echo "timeout reached while waiting for sonar to get healthy"
-    exit 1
-  fi
+  echo "Waiting for SonarQube to get up (max. 120 seconds)..."
+  wait_for_sonar_to_get_up 120
 
-  # Waiting for SonarQube to be started based on the log file
-  # Could not work if old log file is already present, e.g. after upgrade of dogu
-  for i in $(seq 1 10);
-  do
-    # starting compute engine is the last thing sonar does on startup
-    if grep -s "Compute Engine is up" /opt/sonar/logs/sonar.log > /dev/null; then
-      break
-    fi
-    if [ "$i" -eq 10 ] ; then
-      echo "compute engine did not start in the allowed time. Dogu exits now"
-      exit 1
-    fi
-    echo "wait for compute engine to be up"
-    sleep 5
-  done
+  echo "Waiting for SonarQube to get healthy (max. 120 seconds)..."
+  wait_for_sonar_to_get_healthy_with_default_admin_credentials 120
+
+  # import quality profiles if directory is not empty
+  import_quality_profiles_if_present
+
+  echo "setting base url"
+  set_property_with_default_admin_credentials "sonar.core.serverBaseURL" "https://${FQDN}/sonar"
+
+  echo  "adding admin group"
+  create_user_group_with_default_admin_credentials "${ADMIN_GROUP}" "CESAdministratorGroup"
 
-  # import quality profiles
-  import_quality_profiles
-
-  echo "write ces configurations into database"
-	# set base url
-  sql "INSERT INTO properties (prop_key, text_value) VALUES ('sonar.core.serverBaseURL', 'https://${FQDN}/sonar');"
-  # remove default admin
-  sql "DELETE FROM users WHERE login='admin';"
-  # add admin group
-  sql "INSERT INTO groups (name, description, created_at) VALUES ('${ADMINGROUP}', 'CES Administrator Group', now());"
-  # add admin privileges to admin group
-  sql "INSERT INTO group_roles (group_id, role) VALUES((SELECT id FROM groups WHERE name='${ADMINGROUP}'), 'admin');"
-
-  # set email settings
-  sql "INSERT INTO properties (prop_key, text_value) VALUES ('email.smtp_host.secured', 'postfix');"
-  sql "INSERT INTO properties (prop_key, text_value) VALUES ('email.smtp_port.secured', '25');"
-  sql "INSERT INTO properties (prop_key, text_value) VALUES ('email.from', '${MAIL_ADDRESS}');"
-  sql "INSERT INTO properties (prop_key, text_value) VALUES ('email.prefix', '[SONARQUBE]');"
-
-  echo "restarting sonar to account for configuration changes"
-  stopSonar ${SONAR_PROCESS_ID}
+  printf "\\nadding admin privileges to admin group\\n"
+  grant_permission_to_group_with_default_admin_credentials "${ADMIN_GROUP}" "admin"
+
+  echo "setting email configuration"
```

Comment:
Can we do this with our sonar.properties file?

Reply:
Unfortunately not, these settings are not accepted by SonarQube if they are set in the sonar.properties file.

Reply:
EDIT: This configuration is now set via the REST API.